internal/pkg/service/stream/storage/statistics/repository/stats_files_test.go

```diff
@@ -0,0 +1,200 @@
+		sliceKey3 = slices[2].SliceKey
+		sliceKey4 = slices[3].SliceKey
+	}
+
+	// Add some statistics
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		assert.NoError(t, statsRepo.Put(ctx, []statistics.PerSlice{
+			{
+				SliceKey: sliceKey1,
+				Value: statistics.Value{
+					SlicesCount:      1,
+					FirstRecordAt:    utctime.MustParse("2000-01-01T01:00:00.000Z"),
+					LastRecordAt:     utctime.MustParse("2000-01-01T02:00:00.000Z"),
+					RecordsCount:     1,
+					UncompressedSize: 1,
+					CompressedSize:   1,
+				},
+			},
+			{
+				SliceKey: sliceKey2,
+				Value: statistics.Value{
+					SlicesCount:      1,
+					FirstRecordAt:    utctime.MustParse("2000-01-01T02:00:00.000Z"),
+					LastRecordAt:     utctime.MustParse("2000-01-01T03:00:00.000Z"),
+					RecordsCount:     10,
+					UncompressedSize: 10,
+					CompressedSize:   10,
+				},
+			},
+			{
+				SliceKey: sliceKey3,
+				Value: statistics.Value{
+					SlicesCount:      1,
+					FirstRecordAt:    utctime.MustParse("2000-01-01T03:00:00.000Z"),
+					LastRecordAt:     utctime.MustParse("2000-01-01T04:00:00.000Z"),
+					RecordsCount:     100,
+					UncompressedSize: 100,
+					CompressedSize:   100,
+				},
+			},
+			{
+				SliceKey: sliceKey4,
+				Value: statistics.Value{
+					SlicesCount:      1,
+					FirstRecordAt:    utctime.MustParse("2000-01-02T01:00:00.000Z"),
+					LastRecordAt:     utctime.MustParse("2000-01-02T04:00:00.000Z"),
+					RecordsCount:     1,
+					UncompressedSize: 1,
+					CompressedSize:   1,
+				},
+			},
+		}))
+	}
+
+	// Disable sink, it triggers closing of the active file, so all 4 slices are now in the Closing state
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, defRepo.Sink().Disable(sinkKey, clk.Now(), by, "some reason").Do(ctx).Err())
+	}
+
+	// Move slice1 and slice2 to the staging level, so 2 slices are in Uploaded state, 2 in Writing state.
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, sliceRepo.SwitchToUploading(sliceKey1, clk.Now()).Do(ctx).Err())
+		require.NoError(t, sliceRepo.SwitchToUploading(sliceKey2, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, sliceRepo.SwitchToUploaded(sliceKey1, clk.Now()).Do(ctx).Err())
+		require.NoError(t, sliceRepo.SwitchToUploaded(sliceKey2, clk.Now()).Do(ctx).Err())
+	}
+
+	// File stats
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		start := model.FileID{OpenedAt: utctime.MustParse("2000-01-01T00:00:00.000Z")}
+		end := model.FileID{OpenedAt: utctime.MustParse("2000-01-02T00:00:00.000Z")}
+		result, err := statsRepo.FilesStats(sliceKey1.SinkKey, start, end).Do(ctx).ResultOrErr()
```

Comment:
If I understand it correctly, we ask the Sink of slice 1 about the files stats of the `Sink` not of the `slice` only? Start and end represents the statistics boundaries. However in `LastRecordAt` part we can obtain > timestamp than the `end` specifies.

Reply:
> If I understand it correctly, we ask the Sink of slice 1 about the files stats of the Sink not of the slice only? 

I don't understand.

> Start and end represents the statistics boundaries. However in LastRecordAt part we can obtain > timestamp than the end specifies.

Yes.

Reply:
The stats are per `SinkKey` not per `slice` (basically the statics can be withdrawn with single object (file/slice) as both have reference to `SinkKey`). So we can access the statistics both by `sinkKey` or by `sliceKey1.SinkKey`, even `file1.SinkKey`. Am I right?

Reply:
:telephone_receiver:

Reply:
`File stats` -> `Files stats`

Reply:
https://github.com/keboola/keboola-as-code/pull/1820/commits/5cdcbe4bd10c8afa0adb65b4de2bc372c7216f64